hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/BucketEndpoint.java

```diff
@@ -292,6 +294,14 @@ public Response get(
     return Response.ok(response).build();
   }
 
+  private int validateMaxKeys(int maxKeys) throws OS3Exception {
+    if (maxKeys <= 0) {
+      throw S3ErrorTable.newError(S3ErrorTable.INVALID_ARGUMENT, "maxKeys must be > 0");
+    }
+
+    return Math.min(maxKeys, 1000);
```

Comment:
This sever side max needs to be configurable. Depending on deployment scale and resources, a larger number maybe desired for performance.

Reply:
I implement this function but not yet commit.

If below statement is not wrong, should config maxKeys still need? Or for the purpose to reduce maxKeys like limit to 500(some number is smaller than 1000)?


> In AWS S3 ListObjects & ListObjectsV2 API reference, they both mention below.
> 
> > Returns some or all (up to 1,000) of the objects in a bucket.
> 
> Therefore I think add the maxKeys up to 1000 is appropriate.
> 
> reference: [ListObjects](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjects.html) [ListObjectsV2](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html)

Reply:
I agree that AWS has a limit it enforces. For Ozone, we should set the default to 1000, but in some high-performance use cases, it might make sense to have it higher as well, where the listing needs to be completed in one round trip. If we make it configurable, the customer will have a choice.

Reply:
Get it now. I kind of stuck at AWS S3 protocol, but in Ozone we can make it more flexible.

Reply:
Configurable maxKey test result with some local log
![image](https://github.com/user-attachments/assets/cdea7c1f-dfe3-4a73-901a-4077d53d989e)